readium/shared/src/main/java/org/readium/r2/shared/util/http/DefaultHttpClient.kt

```diff
@@ -39,7 +40,11 @@ import timber.log.Timber
 public class DefaultHttpClient(
     private val mediaTypeRetriever: MediaTypeRetriever,
     private val userAgent: String? = null,
-    private val additionalHeaders: Map<String, String> = mapOf(),
+    @Deprecated(
+        "Pass all headers when building your request or modify it in Callback.onStartRequest",
+        level = DeprecationLevel.ERROR
+    )
```

Comment:
Unfortunately we can't use `@Deprecated` on a function parameter. I added a second constructor to deprecate it.